semantic/monotype.go

```diff
@@ -271,17 +276,39 @@ func getArr(tbl fbTabler) (*fbsemantic.Arr, error) {
 	return arr, nil
 }
 
-// ElemType returns the element type if this monotype is an array, and an error otherise.
+func getVec(tbl fbTabler) (*fbsemantic.Vector, error) {
+	vec, ok := tbl.(*fbsemantic.Vector)
+	if !ok {
+		return nil, errors.New(codes.Internal, "MonoType is not a vector")
+	}
+	return vec, nil
+}
+
+// ElemType returns the element type if this monotype is an array or vector, and an error otherise.
 func (mt MonoType) ElemType() (MonoType, error) {
-	arr, err := getArr(mt.tbl)
-	if err != nil {
-		return MonoType{}, err
+	// XXX: sean (16 Dec 2021) - This err1/err2 business seems sloppy. There's
+	// probably a better way to do this. This is functional for now though.
+	// Will revisit later.
+	arr, err1 := getArr(mt.tbl)
+	vec, err2 := getVec(mt.tbl)
+	if err1 != nil && err2 != nil {
+		return MonoType{}, errors.New(codes.Internal, "MonoType is not an array or a vector")
```

Comment:
I am regretting making all of these functions return errors. I think they should just panic, if we try to get the elem type of an int or whatever, that's always an internal error. It would just simplify all this logic so much.